pkg/epp/handlers/server.go

```diff
@@ -377,6 +380,15 @@ func (s *StreamingServer) populateRequestHeaderResponse(reqCtx *RequestContext,
 			},
 		})
 	}
+	// Add headers added by filters/scorers
+	for key, value := range mutatedHeaders {
```

Comment:
The headers is a list. If the plugins updated the value of an existing header key, should we update the header here instead of just appending?